pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+                restake,
+                creator_info.state,
+                staker_info.latest_staked_value(),
+            );
+
+            if should_restake_reward {
+                staker_info
+                    .stake(current_era, staker_reward)
+                    .map_err(|_| Error::<T>::UnexpectedStakeInfoEra)?;
+
+                // Restaking will, in the worst case, remove one, and add one record,
+                // so it's fine if the vector is full
+                ensure!(
+                    staker_info.len() <= T::MaxEraStakeValues::get(),
+                    Error::<T>::TooManyEraStakeValues
+                );
+            }
+
+            // Withdraw reward funds from rewards holding account
+            let reward_imbalance = T::Currency::withdraw(
+                &Self::account_id(),
+                staker_reward,
+                WithdrawReasons::TRANSFER,
+                ExistenceRequirement::AllowDeath,
+            )?;
+
+            if should_restake_reward {
+                ledger.locked = ledger.locked.saturating_add(staker_reward);
+                Self::update_ledger(&staker, ledger);
+
+                // Update storage
+                GeneralEraInfo::<T>::mutate(current_era, |value| {
+                    if let Some(x) = value {
+                        x.staked = x.staked.saturating_add(staker_reward);
+                        x.locked = x.locked.saturating_add(staker_reward);
+                    }
+                });
+
+                CreatorEraStake::<T>::mutate(space_id, current_era, |staking_info| {
+                    if let Some(x) = staking_info {
+                        x.total = x.total.saturating_add(staker_reward);
+                    }
+                });
+
+                Self::deposit_event(Event::<T>::Staked {
+                    who: staker.clone(),
+                    creator: space_id,
+                    era: current_era,
+                    amount: staker_reward,
+                });
+            }
+
+            // TODO: mint tokens to balance before locking in case `should_restake_reward` is true
+            T::Currency::resolve_creating(&staker, reward_imbalance);
+            Self::update_staker_info(&staker, space_id, staker_info);
+
+            // TODO: change event
+            Self::deposit_event(Event::<T>::RewardsClaimed {
+                who: staker,
+                amount: staker_reward,
+            });
+
+            /*Ok(Some(if should_restake_reward {
+                T::WeightInfo::claim_staker_with_restake()
+            } else {
+                T::WeightInfo::claim_staker_without_restake()
+            })
+                .into())*/
+            Ok(().into())
+        }
+
+        #[pallet::call_index(8)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn claim_creator_reward(origin: OriginFor<T>, space_id: SpaceId, era: EraIndex) -> DispatchResultWithPostInfo {
```

Comment:
If it is possible, let's not force a user to specify "era" -> this will improve UX/DX, similar to `claim_staker_reward`

Reply:
It requires a big refactor, in other way it will consume tons of weight.